rust-runtime/aws-smithy-runtime/src/client/orchestrator.rs

```diff
@@ -61,9 +62,16 @@ pub async fn invoke(
         let retry_strategy = cfg.retry_strategy();
         match retry_strategy.should_attempt_initial_request(cfg) {
             // Yes, let's make a request
-            Ok(_) => {}
+            Ok(ShouldAttempt::Yes) => {}
+            // No, this request shouldn't be retried
+            Ok(ShouldAttempt::No) => {
+                todo!("Create a new dispatch error for this case.")
```

Comment:
For this one, I'm not sure if we'll ever hit this case. I'm expecting that when we don't want to send an initial request, it'll always have some "reasoning" message. If we just get a "No" back, then we can't say why we shouldn't attempt the request. I'll add a provisional error message for this, but I think I may mark it as unreachable in a later PR and add a message about how retry strategies should always either return:
- yes
- yes, after a delay
- no, because <ERROR>

and that the "No, for no reason" case is a panic. I'm open to other ideas though.